python/lsst/ts/m2gui/model.py

```diff
@@ -1077,6 +1048,127 @@ def _process_lost_connection(self):
         self.update_system_status("isCrioConnected", False)
         self.update_system_status("isTelemetryActive", False)
 
+    async def start(self):
+        """Transition from the Standby mode to the Diagnostic mode.
+
+        Raises
+        ------
+        `RuntimeError`
+            When the system is not in the Standby mode.
+        """
+
+        if self.local_mode != LocalMode.Standby:
+            raise RuntimeError(
+                f"System is in {self.local_mode!r} instead of {LocalMode.Standby!r}."
+            )
+
+        await self.controller.set_closed_loop_control_mode(ClosedLoopControlMode.Idle)
+        await self.controller.power(PowerType.Communication, True)
+
+        self.local_mode = LocalMode.Diagnostic
+
+        self.report_control_status()
+
+    async def enable(self):
+        """Transition from the Diagnostic mode to the Enable mode.
```

Comment:
I am not sure enabled/disabled states should be misused for diagnostic mode. M1M3 has (I believe) similar problem, yet please look https://ts-m1m3support.lsst.io/dir_633db867517fc930a7645f2dac5f41c1.html how diagnostic (aka Engineering) mode is implemented.

I believe those are hooks for SAL state transition commands the class is acting on. Then disable is meant for the mirror state when it cannot be commanded. Enabled is for states when the mirror can be commanded. The granularity of accepted commands is then ruled by sub-state, which can be parked, EngineeringParked (diagnostic parked), ...

Reply:
As the discussion, renamed the state transition functions in commit: [b10208d](https://github.com/lsst-ts/ts_m2gui/pull/28/commits/b10208d77d2457fe1d385c7f87f4f7a64ec7a531). Thanks!